ivadomed/scripts/automate_training.py

```diff
@@ -123,74 +129,496 @@ def test_worker(config):
+        initial_config (dict): The original config file, which we use as a basis from which
+            to modify our hyperparameters.
+
+            .. code-block:: JSON
+
+                {
+                    "training_parameters": {
+                        "batch_size": 18,
+                        "loss": {"name": "DiceLoss"}
+                    },
+                    "default_model":     {
+                        "name": "Unet",
+                        "dropout_rate": 0.3,
+                        "depth": 3
+                    },
+                    "model_name": "seg_tumor_t2",
+                    "path_output": "./tmp/"
+                }
+        all_combin (bool): If true, combine the hyperparameters combinatorically.
+        multi_params (bool): If true, combine the hyperparameters by index in the list, i.e.
+            all the first elements, then all the second elements, etc.
+
+    Returns:
+        list, dict: A list of configuration dictionaries, modified by the hyperparameters.
+
+        .. code-block:: python
+
+            config_list = [
+                {
+                    "training_parameters": {
+                        "batch_size": 18,
+                        "loss": {"name": "DiceLoss"}
+                    },
+                    "default_model":     {
+                        "name": "Unet",
+                        "dropout_rate": 0.3,
+                        "depth": 3
+                    },
+                    "model_name": "seg_tumor_t2",
+                    "path_output": "./tmp/-loss={'name': 'DiceLoss'}"
+                },
+                {
+                    "training_parameters": {
+                        "batch_size": 18,
+                        "loss": {"name": "FocalLoss", "gamma": 0.2, "alpha": 0.5}
+                    },
+                    "default_model":     {
+                        "name": "Unet",
+                        "dropout_rate": 0.3,
+                        "depth": 3
+                    },
+                    "model_name": "seg_tumor_t2",
+                    "path_output": "./tmp/-loss={'name': 'FocalLoss', 'gamma': 0.2, 'alpha': 0.5}"
+                },
+                # etc...
+            ]
+
+    """
+    config_list = []
+    if all_combin:
+        keys = set([hyper_option.base_key for hyper_option in param_list])
+        for combination in list(itertools.combinations(param_list, len(keys))):
+            if keys_are_unique(combination):
+                new_config = copy.deepcopy(initial_config)
+                path_output = new_config["path_output"]
+                for hyper_option in combination:
+                    new_config = update_dict(new_config, hyper_option.option, hyper_option.base_key)
+                    path_output = path_output + hyper_option.name
+                new_config["path_output"] = path_output
+                config_list.append(new_config)
```

Comment:
This block of code is very similar to lines 242 - 248. I would suggest maybe refactoring some of this logic into a helper function to reduce code duplication